lib/src/layer/tile_layer/tile_painter.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_map/src/layer/tile_layer/tile_model.dart';
+
+/// Draws [TileModel]s onto a canvas at the correct position
+class TilePainter extends CustomPainter {
+  /// List of [TileModel]s to draw to the canvas
+  List<TileModel> tiles;
+
+  /// Paint to use when drawing each tile
+  ///
+  /// Defaults to [defaultTilePaint].
+  final Paint? tilePaint;
+
+  /// Default [tilePaint]er
+  static Paint get defaultTilePaint => Paint()
```

Comment:
Can you explain these choices?

If the input tile size is equal to the output tile size, which should be always AFAIK, then no scaling should ever be applied, as far as I understand. Therefore, there should be no reason to use `FilterQuality`, and no anti-aliasing required I think.

I'm not entirely sure though, just wanted to get a good explanation of why these were chosen.

Reply:
What I understood is that when zooming in or out it scales current tile until new tiles are loaded, that is why need to scale image to fit that tile. Initially I did not set filterQuality and had very low quality image. Setting anti-aliasing to false removes ghost lines between tiles.

Reply:
Ah, of course, I forgot about zooming! OK, `FilterQuality` makes sense then.

In regards to the thin lines, we used to have issues with those without canvas, because we weren't careful enough when it came to rounding operations. May be worth double checking any positioning/rounding maths. But I suppose it makes sense for now.